### PR TITLE
issue/2 additional display configurations

### PR DESCRIPTION
**Pull request by kirsty-hames**
- option to set mobile specific image src (e.g display a smaller logo for mobile view)
- option to hide logo for mobile view (declutter the navigation bar for mobile view where limited space is available)
- AT friendly pre-set display options for logo (pre-set includes `small`, `large` or `fill`. As these are applied as classes it gives scope for custom classes in FW projects however the pre-set should cover most use cases)
- documentation updated

### CODE REVIEW

File: js/LogoView.js [hunk 1]
```diff
@@ -8,11 +8,47 @@ define([
...
+        this.setLogoImageSrc();
+      }
+    },
+
+    hideLogoForMobile: function() {
+      (Adapt.device.screenSize === 'small')
+        // mobile logo display
+        ? $(".navigation-logo__image").addClass('u-display-none')
+        // desktop logo display
+        : $(".navigation-logo__image").removeClass('u-display-none');
```
Comment (tomgreenfield) [inline js/LogoView.js:33]: Suggest assigning ternary to variable e.g.
```js
var isDeviceSmall = Adapt.device.screenSize === 'small';

$(".navigation-logo__image").toggleClass('u-display-none', isDeviceSmall);
```

File: js/LogoView.js [hunk 2]
```diff
@@ -8,11 +8,47 @@ define([
...
+
+    setLogoImageSrc: function() {
+      var config = Adapt.course.get("_navigationLogo");
+      var mobileSrc = config._graphic._mobileSrc;
+      var src = config._graphic._src;
+      (Adapt.device.screenSize === 'small')
+        // mobile logo display
+        ? $(".navigation-logo__image").attr('src', mobileSrc)
+        // desktop logo display
+        : $(".navigation-logo__image").attr('src', src);
```
Comment (tomgreenfield) [inline js/LogoView.js:44]: Suggest assigning ternary to variable, e.g.
```js
var config = Adapt.course.get('_navigationLogo')._graphic;
var src = Adapt.device.screenSize === 'small' ?  config._mobileSrc : config._src;

$('.navigation-logo__image').attr('src', src);
```
You could also pass in `config` as a parameter from `postRender()`, negating the repeated look-up.

File: properties.schema [hunk 3]
```diff
@@ -45,10 +54,26 @@
...
                     }
                   }
+                },
+                "_logoSize": {
+                  "type": "string",
+                  "required": false,
+                  "default": "large",
+                  "inputType": {"type":"Select", "options":["small","large","fill"]},
+                  "title": "Set the size of the logo image",
+                  "help": "Small: Logo image displays at the same size as navigation icons. Large: Logo image displays larger than navigation icons leaving small space arounf the logo. Fill: Logo image fills the height of the navigation bar."
```
Comment (tomgreenfield) [inline properties.schema:68]: Typo: `arounf`.

**Comment by guywillis** [comment]
@kirsty-hames Would you mind reviewing my amends please?

File: bower.json [hunk 4]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "adapt-navigationLogo",
-  "version": "2.0.0",
+  "version": "2.2.0",
```
Comment (tomgreenfield) [inline bower.json:3]: Meant to jump from 2 to 2.2?

Comment (guywillis) [reply bower.json]: Nope, good spot

**Comment by kirsty-hames** [comment]
> As discussed, how about simplifying `_logoSize` to a simple `_fill` attribute to be set to `true` or `false`? The small option feels too rare to be useful.

I'd agree and this would simplify the config. @kineojen, @guywillis do you have any objections to this?

**Comment by kineojen** [comment]
> > As discussed, how about simplifying `_logoSize` to a simple `_fill` attribute to be set to `true` or `false`? The small option feels too rare to be useful.
> 
> I'd agree and this would simplify the config. @kineojen, @guywillis do you have any objections to this?

Nope, sounds good.

File: properties.schema [hunk 5]
```diff
@@ -19,24 +19,34 @@
                 "_isEnabled": {
                   "type": "boolean",
                   "required": false,
-                  "title": "Enable Navigation Logo",
+                  "default": false,
+                  "title": "Enable Navigation Logo?",
```
Comment (tomgreenfield) [inline properties.schema:23]:
```suggestion
                  "title": "Add logo to navigation bar",
```

File: properties.schema [hunk 6]
```diff
@@ -19,24 +19,34 @@
...
                   "inputType": "Checkbox",
                   "validators": [],
                   "help": "Set to true to enable the 'Navigation Logo' feature."
```
Comment (tomgreenfield) [inline properties.schema:26]:
```suggestion
                  "help": ""
```

File: properties.schema [hunk 7]
```diff
@@ -45,10 +55,28 @@
                       "title": "Logo image alternative Text",
```
Comment (tomgreenfield) [inline properties.schema:55]:
```suggestion
                      "title": "Logo image alternative text",
```

File: properties.schema [hunk 8]
```diff
@@ -45,10 +55,28 @@
...
                       "inputType": "Text",
                       "validators": [],
-                      "help": "The alternative text for this image",
+                      "help": "The alternative text for this image.",
```
Comment (tomgreenfield) [inline properties.schema:58]: I'd instead mention something about screenreaders here – at the moment it's just a repeat of the title.

File: properties.schema [hunk 9]
```diff
@@ -45,10 +55,28 @@
...
                       "translatable": true
                     }
                   }
+                },
+                "_fillNavHeight": {
+                  "type": "boolean",
+                  "required": false,
+                  "default": false,
+                  "title": "Logo fills the nav bar height",
```
Comment (tomgreenfield) [inline properties.schema:67]:
```suggestion
                  "title": "Make logo fill navigation bar height",
```

File: properties.schema [hunk 10]
```diff
@@ -45,10 +55,28 @@
...
+                  "inputType": "Checkbox",
+                  "validators": [],
+                  "help": "Logo displays with minimal padding by default. Set to true for logo to fill the nav bar height."
+                },
+                "_hideLogoForMobile": {
+                  "type": "boolean",
+                  "required": false,
+                  "default": false,
+                  "title": "Hide Logo for mobile view",
```
Comment (tomgreenfield) [inline properties.schema:76]:
```suggestion
                  "title": "Hide logo for mobile view",
```

File: README.md [hunk 11]
```diff
@@ -1,4 +1,40 @@
 # adapt-navigationLogo
 
-AT Compatible
-Accessible
+Displays an image in the course navigation bar. Image can be configured to utilise three different sizings (small, large, and fill), whether the image should display on small screens, and if the image should be shown on a small screen whether an alternative, more mobile friendly, version should be used.
```
Comment (tomgreenfield) [inline README.md:3]: Description needs updating – it still mentions sizings.

File: properties.schema (hunk 8 above)
Comment (kirsty-hames) [reply properties.schema:58]: Thanks @tomgreenfield, a bit wordy but how about the following.

Provide alternative text for the image, such as your company name. This text is not a visible element. It is utilized by assistive technology such as screen readers. Avoid identifying the logo as actually being a logo.

Comment (tomgreenfield) [reply properties.schema:58]: 🤯 superb.

File: README.md (hunk 11 above)
Comment (kirsty-hames) [reply README.md]: All amends are done. If you're happy with the last two commits @tomgreenfield I'll merge the PR.
